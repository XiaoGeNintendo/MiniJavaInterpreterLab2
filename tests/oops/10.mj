class A{
    string t;
    A(){
        this(1);
        t+="a1";
    }

    A(int x){
        this("asd");
        t+="a2";
    }

    A(string x){
        t+="a3";
    }
}

class B extends A{
    B(){
        this(1);
        t+="b1";
    }

    B(int x){
        this("asd");
        t+="b2";
    }

    B(string x){
        t+="b3";
    }
}

class C extends B{
    C(){
        this(1);
        t+="c1";
    }

    C(int x){
        this("asd");
        t+="c2";
    }

    C(string x){
        super(3);
        t+="c3";
    }
}

int main(){
    println(new C().t);
    return 0;
}