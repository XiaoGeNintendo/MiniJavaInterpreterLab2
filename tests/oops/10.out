a3a2a1b3b2c3c2c1
Process exits with the code 0.